kbs/docs/initdata.md

```diff
@@ -0,0 +1,267 @@
+digest of the `data`` part in the Initdata Metadata to the hardware TEE-specific 
+field in evidence. This field will be checked by the verifier during the remote
+attestation. This will be introduced in 
+[Integrity Binding for Different TEEs](#integrity-binding-for-different-tees)
+- A data serialization method. This method is used to serialize the Initdata Metadata
+into a canonicalized one, which will help to get consistent cryptographic hash for
+different forms of a same Initdata Metadata. This will be introduced in
+[Data Canonicalization Algorithm](#data-canonicalization-algorithm).
+
+This spec does not define how the Initdata Metadata will be delivered into the TEE.
+Different projects will have its own way to do this. For Confidential Containers,
+we will use kata-runtime and kata-agent to collaborate to achieve this function.
+
+## Terminology
+
+This section will introduce the terminology used in this spec to avoid ambiguity.
+
+- `Initdata`: Data that needs to be injected when the TEE is started. This data
+requires integrity protection but does not require confidentiality protection.
+- `Initdata Metadata`: A data structure that includes initdata and other information
+that will help to calculate the initdata digest.
+- `Initdata digest`: Digest of the initdata. It will be used as the value of the
+TEE HOSTDATA field.
+- `HOSTDATA`: Fields that can be bound to a specific TEE instance. This field
+information will be included in the TEE-signed remote attestation report. Typically,
+Intel TDX's `mr_config_id`, AMD SNP's `hostdata` and Arm CCA's `CCA_REALM_PERSONALIZATION_VALUE`.
+In order to avoid confusion with the hostdata field of AMD SNP, when we do not
+emphasize a specific platform s.t. SNP, we are referring to the corresponding fields of
+various TEE platforms.
+
+
+## Specifications
+
+### Initdata Metadata
+
+Initdata Metadata defines a standardized initdata data format. Please note that it
+does not indicate the specific encoding format, but requires that the encoding format
+must support the expression of key-value data pairs. Typical encodings that meet
+this requirement include JSON, TOML and YAML, etc.
+
+An Initdata Metadata SHOULD have the following fields
+- `version`: The format version of the initdata metadata. Version number will provide
+extensibility. The definition in this spec is all `0.1.0`.
+- `algorithm`: The hash algorithm to calculate the value to set as `HOSTDATA`. The typical
+algorithms are `sha256`, `sha384`, `sha512`.
+- `data`: a key-value map. Including the concrete content of initdata. Nested maps are
+also supported.
+- `digest`: the digest of the canonicalized `data` field using hash algorithm specified by
+`algorithm`. Note that the `digest` itself is not integrity protected in the current
+specification, so it may be tampered with. Therefore, this field itself can only be used as
+a reference. It is **NOT** recommended to rely on this field in the software stack in a specific TEE.
```

Comment:
In fact, technically this field is really useless because it can be tampered with.

But imagine a scenario where the user has a lot of initdata metadata. He may also configure the attestation policy and fill in the reference value of the corresponding initdata field. At this time, he can copy and paste the `digest` directly instead of storing an additional copy for each one.

I am not sure if it makes sense, but I am ok to either
1. deprecate the field
2. leave it as-is and emphasize this field will be tampered by untrust host in the doc

Reply:
IMHO, if we want the `digest` to serve as a reference and prevent users from relying on the field's value rather than calculating the digest themselves, we should generate the value using a different algorithm, not the hash algorithm specified by the algorithm parameter.

Reply:
I guess I can see the purpose of having the digest as a reference, but it seems like the verifier could just calculate the digest for this purpose. It's going to calculate the digest anyway to verify the binding to the evidence.

Reply:
I prefer to mark this field as untrusted in the doc and still keep it as-is. The reason is for a concrete usage mentioned in https://github.com/confidential-containers/trustee/pull/348#discussion_r1517081105

Also, if we use a different hash algorithm upon the field, this may be useful, but it seems to have no specific help for the consumption and production process of initdata currently.

Reply:
If it's yaml or toml, such a hash could be comment, I guess. I'm not convinced think we need to include it into the schema for copy paste operations.

Reply:
Ok. Since this would bring a lot of troubles, let me delete the field.